### PR TITLE
Code cleanup, optimisations and lint

### LINKED ISSUE #21: Improve readability of webhook URLs creation
Suggestion for a better query map builder:

```js
// Have this helper
const urlBuilder = (url, queries) => {
  const params = new URLSearchParams();
  Object.entries(queries).forEach(([key, value]) => params.append(key, value));
  return `${url}?${params}`;
}

// then your `action` here can be
const queries = {
  mode: 'mainProcess',
  callsid: callSid,
  cbphone: explodePhone('format', phone),
};
if (event.taskSid) {
  queries.taskSid = event.taskSid;
}
const action = urlBuilder(`${domain}/inqueue-callback`, queries);
```

Easier to read IMO

_Originally posted by @ktalebian in https://github.com/twilio-labs/plugin-queued-callbacks-and-voicemail/pull/20#discussion_r660964928_

### PR DESCRIPTION
**Pull request by vernig**
The aim of this PR is to refactor the solution backend code (Twilio Serverless) and clean it up of unused functions, routines and variables. 

The main additions in this PR are:

* public/resources/functions/helpers.private.js: this file contains helpers functions used by the other endpoints
* public/resources/functions/options.private.js: this file contains a single object with all the options used by the solution backed

This PR also:
Fix #18
Fix #21

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Comment by vernig** [comment]
@ktalebian I rebased my PR on #24. 

I disabled some of the `eslint` in the serverless, namely: 
* `camelcase`: this requires change in the plugin. We can do that in a follow-up PR
* `sonarjs/cognitive-complexity`: this needs a bigger refactoring (and more time)

We need more end-to-end testing before merging (cc @bermudezmt )

**Comment by vernig** [comment]
@ktalebian @bermudezmt I've done some end-to-end testing and fix some bugs. The code is ready for review / testing.

File: public/resources/functions/helpers.private.js
Comment (ktalebian) [inline public/resources/functions/helpers.private.js:40]: Redundant - `return { ... }`

File: public/resources/functions/helpers.private.js
Comment (ktalebian) [inline public/resources/functions/helpers.private.js:54]: Same here

File: public/resources/functions/helpers.private.js
Comment (ktalebian) [inline public/resources/functions/helpers.private.js:81]: Can you convert these two `camelCase`?

File: public/resources/functions/helpers.private.js
Comment (ktalebian) [inline public/resources/functions/helpers.private.js:64]: Would be nice to have a js docblock for these methods as well!

File: public/resources/functions/inqueue-utils.js
Comment (ktalebian) [inline public/resources/functions/inqueue-utils.js:147]: better to call these different variables? 

```js
const task = await getTask(event.taskSid);
const updatedTask = await updateTask(event.taskSid, task.attr);
```

File: public/resources/functions/inqueue-utils.js
Comment (ktalebian) [inline public/resources/functions/inqueue-utils.js:204]: Cleaner to use spread operator:

```js
newAttributes = { ...newAttributes, { conversations: { conversation_id: event.taskSid } } };
```

**Review by bermudezmt — CHANGES_REQUESTED** [review]
Tested the queued callback and all was working as expected. Testing voicemail however led to a Function execution error (Error - 82005):

```
Function execution resulted in an error log: ETIMEDOUT | stack: Error: ETIMEDOUT
    at Timeout.<anonymous> (/var/task/node_modules/request/request.js:848:19)
    at listOnTimeout (internal/timers.js:554:17)
    at processTimers (internal/timers.js:497:7)
 ```
 @vernig Since I added you to that account, maybe you can review the logs in the debugger.

File: public/resources/functions/helpers.private.js
Comment (vernig) [reply public/resources/functions/helpers.private.js:81]: I'll do it in a follow-up PR @ktalebian because they require changes in the plugin code as well. I want to keep this PR only for serverless

File: public/resources/functions/helpers.private.js
Comment (vernig) [reply public/resources/functions/helpers.private.js]: Solved in e98d19b

File: public/resources/functions/helpers.private.js
Comment (vernig) [reply public/resources/functions/helpers.private.js:70]: Solved in e98d19b

File: public/resources/functions/helpers.private.js
Comment (vernig) [reply public/resources/functions/helpers.private.js]: Solved in e98d19b

File: public/resources/functions/helpers.private.js
Comment (vernig) [reply public/resources/functions/helpers.private.js:94]: Raised #28 to track this

File: public/resources/functions/inqueue-utils.js
Comment (vernig) [reply public/resources/functions/inqueue-utils.js]: Solved in e98d19b

**Comment by vernig** [comment]
Thanks again @bermudezmt for testing the code. I actually found a bug (introduced when I cleaned up the code). It's now fixed in d68ed62. Thanks again for spotting it!

**Review by bermudezmt — APPROVED** [review]
LGTM after the follow-up commits. @ktalebian feel free to merge if you're satisfied with the changes.

File: public/resources/functions/helpers.private.js
Comment (ktalebian) [inline public/resources/functions/helpers.private.js:89]: Nitpick: remove this

**Review by ktalebian — APPROVED** [review]

File: public/resources/functions/inqueue-utils.js
Comment (vernig) [reply public/resources/functions/inqueue-utils.js]: Solved in e98d19b

File: public/resources/functions/helpers.private.js
Comment (vernig) [reply public/resources/functions/helpers.private.js]: Solved in ab52287

**Comment by vernig** [comment]
@rnairtwilio do you want to have a look to the PR before I merge it?

**Pull request merged**
